src/components/transport_manager/src/transport_adapter/transport_adapter_impl.cc

```diff
@@ -776,6 +776,25 @@ std::string TransportAdapterImpl::GetConnectionType() const {
   return devicesType[GetDeviceType()];
 }
 
+bool TransportAdapterImpl::IsConnectionFinished(
+    const DeviceUID& device_handle, const ApplicationHandle& app_handle) const {
+  bool is_finished = true;
+  LOG4CXX_TRACE(
+      logger_,
+      "check if connection is finalized. device_id: " << &device_handle << ", app_handle: " << &app_handle);
+  connections_lock_.AcquireForReading();
+  ConnectionMap::const_iterator it = connections_.find(
+      std::make_pair(device_handle, app_handle));
+  if (it != connections_.end()) {
+    const ConnectionInfo& info = it->second;
+    if (info.state != ConnectionInfo::FINALISING){
```

Comment:
Please, run [cpplint](https://adc.luxoft.com/confluence/display/APPLINK/2015/10/07/Code+verification+and+beatifying) for avoiding googlestyle-related comments ;)